game-core/src/main/java/games/strategy/engine/framework/map/download/DownloadRunnable.java

```diff
@@ -1,65 +1,62 @@
 package games.strategy.engine.framework.map.download;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
-import java.io.File;
 import java.io.IOException;
+import java.io.InputStream;
 import java.nio.file.Files;
-import java.util.ArrayList;
+import java.nio.file.Path;
+import java.util.Collections;
 import java.util.List;
-import java.util.Optional;
 import java.util.logging.Level;
 
-import games.strategy.engine.ClientFileSystemHelper;
-import games.strategy.io.IoUtils;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import games.strategy.engine.framework.system.HttpProxy;
 import lombok.extern.java.Log;
 
 /**
  * Downloads a map index file, parses it and returns a <code>List</code> of <code>DownloadFileDescription</code>.
  */
 @Log
 public class DownloadRunnable {
-  private final String urlString;
 
-  public DownloadRunnable(final String urlString) {
-    this.urlString = urlString;
-  }
+  private DownloadRunnable() {}
 
   /**
-   * Returns a parsed list of parsed downloadable maps. If initialized with a URL then we will do a network fetch and
-   * parse those contents, otherwise (for testing) we assume a local file reference and parse that.
+   * Parses a file at the given URL into a List of {@link DownloadFileDescription}s.
+   * If an error occurs this will return an empty list.
    */
-  public Optional<List<DownloadFileDescription>> getDownloads() {
-    return beginsWithHttpProtocol(urlString) ? downloadFile() : Optional.of(readLocalFile());
-  }
-
-  private static boolean beginsWithHttpProtocol(final String urlString) {
-    return urlString.startsWith("http://") || urlString.startsWith("https://");
-  }
-
-  private Optional<List<DownloadFileDescription>> downloadFile() {
-    try {
-      final File tempFile = ClientFileSystemHelper.newTempFile();
-      tempFile.deleteOnExit();
-      DownloadConfiguration.contentReader().downloadToFile(urlString, tempFile);
-      final byte[] contents = Files.readAllBytes(tempFile.toPath());
-      return Optional.of(IoUtils.readFromMemory(contents, DownloadFileParser::parse));
+  public static List<DownloadFileDescription> download(final String url) {
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+      final HttpGet request = new HttpGet(url);
+      HttpProxy.addProxy(request);
+      try (CloseableHttpResponse response = client.execute(request)) {
+        final int status = response.getStatusLine().getStatusCode();
+        if (status != HttpStatus.SC_OK) {
+          log.warning("Invalid map link '" + url + "'. Server returned " + status);
+          return Collections.emptyList();
+        }
+        return DownloadFileParser.parse(response.getEntity().getContent());
```

Comment:
This is kind-of duplicating code in `ContentReader.java`, however this class is not suited for operations that don't involve an outputstream.
So in order to re-use this code, the `ContentReader` class needs refactoring first.

Reply:
Would a `TODO: move networking code into ContentReader` help to encourage the next person who comes across this code to do the refactoring? :smile: